core/src/main/java/com/amplifyframework/Platform.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework;
```

Comment:
Is this the right package? Is there a more specific one you can use? What about `.core`, at least?

Or what about moving to an inner enum of the `AmplifyConfiguration`?

Reply:
I didn't want to move it as an inner enum, since that makes the user need to specify `AmplifyConfiguration.Platform.FLUTTER` instead of just `Platform.FLUTTER`. It also keeps it more hidden from plain sight.

Also, `UserAgent` is currently configured globally with `Map<Platform, String>`. Making `Platform` a part of `AmplifyConfiguration` would couple it with `UserAgent`, which you mentioned you would rather not do.

Reply:
As for whether it belongs inside `core` package or the outermost package, I'm not sure what the best approach would be. Personally, I thought that `Platform` specification is one of the most broad concepts in the library, so it might as well belong outside of the core package. I don't think there's a big distinction between `com.amplifyframework` and `com.amplifyframework.core` packages as of right now.